Website MVP

tremainebuchanan opened the pull request:
The pull request represents the website's initial launch. 
The front-end in broken down into blocks/modules. Each module is composed of three files namely: an html file representing the view, the module file defining the dependencies, routes and other module configurations and the module controller file.

 There are four main modules: 
- Home - renders the home page of the application. 
- Page -  renders a page created within Keystone.
- Post - renders a post created within Keystone.
- Shared Services - defines application wide services such as fetching data from the back end.
- Filter - defines services which handles modification of data for rendering.

tremainebuchanan committed:
Completed post endpoint

tremainebuchanan committed:
Configured front end with backend

tremainebuchanan committed:
Post page completed

tremainebuchanan committed:
MVP Complete

tremainebuchanan committed:
Added public/bower_components to gitignore

tremainebuchanan committed:
Bug fix

tremainebuchanan committed:
Aligned develop and feature branch gitignore files

matjames007 commented on public/app/modules/home/home.html:
```diff
@@ -10,129 +10,103 @@
+								<div class="label label-text-2 black-text">IDEAS</div>
                                     <span class="title-2 white-text">
-                                        A Caribbean Open Data Public Service
+                                        {{page_content[0].title}}
```
@tremainebuchanan u know how i feel about the use of `[0]` in code....

matjames007 commented on public/app/modules/home/home.html:
```diff
@@ -10,129 +10,103 @@
-                                        A Caribbean Open Data Public Service
-                                    </span>
+				</a>
+				<a ng-href="#/{{page_content[1].link}}" ng-if="page_content[1]">
```
@tremainebuchanan format the data in `page_content` so that you don't have to statically reference items.

matjames007 commented on public/app/modules/home/home.html:
```diff
@@ -10,129 +10,103 @@
-                            <span class="title-2 white-text">
-                                A Caribbean Open Data Public Service
-                            </span>
+		<a ng-href="#/{{page_content[2].link}}" ng-if="page_content[2]">
```
@tremainebuchanan or use `ng-repeat` since this thing seems to have a pattern

matjames007 commented on public/app/modules/post/post.html:
```diff
@@ -0,0 +1,38 @@
+			<div class="row">
+				<div class="article no-padding col-sm-8 col-md-9 col-lg-9">
+					<div class="label label-text-2 black-text">
+						IDEAS
```
@tremainebuchanan should this be fixed to `IDEAS`?  Probably should replace using category/post type?

tremainebuchanan replied:
@matjames007 I believe this is an artifact of the design process. We will have a discussion around what should be used here.

matjames007 commented on public/app/modules/post/post.html:
```diff
@@ -0,0 +1,38 @@
+						<div class="box-caption">
+							<div class="label label-text-2 dark-grey-text">Written By</div>
+                                    <span class="box-caption-headline title-2 white-text">
+                                        {{post.author.name.first}} {{post.author.name.last}}
```
@tremainebuchanan might be useful in the future to make the text `{{post.author.name.first}} {{post.author.name.last}}` a link to the profile of the author.

tremainebuchanan replied:
@matjames007 that's the idea. Once the design for the profile page is complete this feature should be completed.

matjames007 commented on public/bower_components/angular-bootstrap/.bower.json:
```diff
@@ -26,7 +26,6 @@
     "commit": "c4d2e80dbbe3df24e0c72d84dda34a35b0a81953"
```
@tremainebuchanan exclude all `bower_components` files and folders from the repo.  I believe the command is:

```
git rm -r  --cached bower_components/
```

matjames007 commented:
@tremainebuchanan overall I really like this.  I guess it is still a work in progress so I will keep looking out for the changes.

tremainebuchanan committed:
Array index reference in home page removed.
The front page content was split in blocks.
Each content group consists three blocks, two
small, one large. The idea was to iterate over an array
containing objects corresponding with the two small
blocks. The large block's content was passed to a variable
and made available to the view.

tremainebuchanan committed:
Front end test suite added

matjames007 commented:
@tremainebuchanan I don't think there would be a big problem with the slicing... but it requires proper documentation to indicate that the position of the post item is dependent on its position in the array.  There should be an emphasis on how a user would publish an item in order for it to be published in a certain position.

tremainebuchanan committed:
Test Suite Added
Karma test runner and Jasmine unit testing implemented in
project.
